src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2096,25 +2134,38 @@ private MethodSymbol InferMethod(BoundCall node, MethodSymbol method, ImmutableA
             // PROTOTYPE(NullableReferenceTypes): Support field initializers in local functions.
         }
 
-        private static BoundExpression RemoveImplicitConversions(BoundExpression expr)
+        private static (BoundExpression, Conversion) RemoveConversion(BoundExpression expr, bool includeExplicitConversions)
         {
-            // PROTOTYPE(NullableReferenceTypes): The loop is necessary to handle
-            // implicit conversions that have multiple parts (for instance, user-defined
-            // conversions). Should check for those cases explicitly.
+            ConversionGroup group = null;
             while (true)
             {
                 if (expr.Kind != BoundKind.Conversion)
                 {
                     break;
                 }
                 var conversion = (BoundConversion)expr;
-                if (conversion.ExplicitCastInCode || !conversion.Conversion.Exists)
+                if (group != conversion.ConversionGroup && group != null)
                 {
                     break;
```

Comment:
What's a situation for this? A cast of a cast, maybe?

Reply:
Yes. Added comment and test.

---
In reply to: [189336566](https://github.com/dotnet/roslyn/pull/26356#discussion_r189336566) [](ancestors = 189336566)